Source/SwiftLintFramework/Rules/VerticalWhitespaceOpeningBracesRule.swift

```diff
@@ -0,0 +1,80 @@
+import Foundation
+import SourceKittenFramework
+
+private extension File {
+    func violatingRanges(for pattern: String) -> [NSRange] {
+        return match(pattern: pattern, excludingSyntaxKinds: SyntaxKind.commentAndStringKinds)
+    }
+}
+
+public struct VerticalWhitespaceOpeningBracesRule {
+    public init() {}
+
+    private static let violatingToValidExamples: [String: String] = [
+        "if x == 5 {\n\n    print(\"x is 5\")": "if x == 5 {\n    print(\"x is 5\")",
+        "if x == 5 {\n\n\n    print(\"x is 5\")": "if x == 5 {\n    print(\"x is 5\")",
+        "if x == 5 {\n\n  print(\"x is 5\")": "if x == 5 {\n  print(\"x is 5\")",
+        "if x == 5 {\n\n\tprint(\"x is 5\")": "if x == 5 {\n\tprint(\"x is 5\")",
+        "struct MyStruct {\n\n    let x = 5": "struct MyStruct {\n    let x = 5",
+        "struct MyStruct {\n\n  let x = 5": "struct MyStruct {\n  let x = 5",
+        "struct MyStruct {\n\n\tlet x = 5": "struct MyStruct {\n\tlet x = 5"
+    ]
+
+    private let pattern = "([{(\\[][ \\t]*)((?:\\n[ \\t]*)+)(\\n)"
+}
+
+extension VerticalWhitespaceOpeningBracesRule: OptInRule {
+    public var configurationDescription: String { return "N/A" }
+
+    public init(configuration: Any) throws {}
+
+    public static let description = RuleDescription(
+        identifier: "vertical_whitespace_opening_braces",
+        name: "Vertical Whitespace after Opening Braces",
+        description: "Don't include vertical whitespace (empty line) after opening braces.",
+        kind: .style,
+        nonTriggeringExamples: Array(Set(violatingToValidExamples.values)),
+        triggeringExamples: Array(Set(violatingToValidExamples.keys)),
+        corrections: violatingToValidExamples
+    )
+
+    public func validate(file: File) -> [StyleViolation] {
+        return file.violatingRanges(for: pattern).map {
+            StyleViolation(
+                ruleDescription: type(of: self).description,
+                severity: ViolationSeverity.warning,
```

Comment:
Most rules allow setting the severity via yml file. One might want to set it error instead of warning, for instance.
To implement that, the rule has to conform to `ConfigurationProviderRule` protocol.

Reply:
Thanks for the suggestion, I made both rules severity configurable.